core/services/gas/block_history_estimator.go

```diff
@@ -424,10 +536,33 @@ func (b *BlockHistoryEstimator) EffectiveGasPrice(block Block, tx Transaction) *
 			priorityFeePerGas = maxFeeMinusBaseFee
 		}
 
-		gasPrice := big.NewInt(0).Add(priorityFeePerGas, block.BaseFeePerGas)
-		return gasPrice
+		effectiveGasPrice := big.NewInt(0).Add(priorityFeePerGas, block.BaseFeePerGas)
+		return effectiveGasPrice
 	default:
-		b.logger.Warnw(fmt.Sprintf("BlockHistoryEstimator: unknown transaction type %v, falling back to gasPrice", tx.Type), "block", block, "tx", tx)
-		return tx.GasPrice
+		b.logger.Warnw(fmt.Sprintf("BlockHistoryEstimator: ignoring unknown transaction type %v", tx.Type), "block", block, "tx", tx)
+		return nil
+	}
+}
+
+func (b *BlockHistoryEstimator) EffectiveTipCap(block Block, tx Transaction) *big.Int {
+	switch tx.Type {
+	case 0x2:
+		return tx.MaxPriorityFeePerGas
+	case 0x0, 0x1:
```

Comment:
Should this case be limited to `0x0`, like  in `bumpGas()`?

Would it be helpful to define some constants for these types? Might be more readable in some places, but repetitive in others. Maybe better in the future with additional types. :shrug:

Reply:
It should not be limited like `bumpGas` because bumpGas controls transactions that we send, and we never send `0x1`.

However, `0x1` transactions _can_ appear on-chain and we should handle those correctly when using the tx for estimation.

I thought about defining constants but it seems unwieldy when its really simple to just type `0x0` or `0x2` etc and everyone that's familiar with Ethereum will know what that means. Seems to be more clear just to be explicit.